hs_core/models.py

```diff
@@ -1642,28 +1640,46 @@ def can_be_public_or_discoverable(self):
 
         return False
 
+    # TODO: these should be instance methods, not class methods.
+    # These decisions are used when operating upon an instance.
+    # That requires less overhead than using a class method.
     @classmethod
     def get_supported_upload_file_types(cls):
```

Comment:
@alvacouch @pkdash I think these methods are called before a resource is created but after a resource type is selected from drop down list for use by file browser. So it sounds to me class method is accurate rather than instance method.